libraries/Microsoft.Bot.Builder/BotFrameworkStreamingExtensionsAdapter.cs

```diff
@@ -0,0 +1,695 @@
+        public async Task<InvokeResponse> ProcessActivityAsync(Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            BotAssert.ActivityNotNull(activity);
+
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            if (cancellationToken == null)
+            {
+                throw new ArgumentNullException(nameof(cancellationToken));
+            }
+
+            _logger.LogInformation($"Received an incoming activity.  ActivityId: {activity.Id}");
+
+            using (var context = new TurnContext(this, activity))
+            {
+                await RunPipelineAsync(context, callback, cancellationToken).ConfigureAwait(false);
+
+                if (activity.Type == ActivityTypes.Invoke)
+                {
+                    var activityInvokeResponse = context.TurnState.Get<Activity>(InvokeReponseKey);
+                    if (activityInvokeResponse == null)
+                    {
+                        return new InvokeResponse { Status = (int)HttpStatusCode.NotImplemented };
+                    }
+                    else
+                    {
+                        return (InvokeResponse)activityInvokeResponse.Value;
+                    }
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Sends activities to the conversation.
+        /// Throws <see cref="ArgumentNullException"/> on null arguments.
+        /// Throws <see cref="ArgumentException"/> if activities length is zero.
+        /// </summary>
+        /// <param name="turnContext">The context object for the turn.</param>
+        /// <param name="activities">The activities to send.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        /// <remarks>If the activities are successfully sent, the task result contains
+        /// an array of <see cref="ResourceResponse"/> objects containing the IDs that
+        /// the receiving channel assigned to the activities.</remarks>
+        /// <seealso cref="ITurnContext.OnSendActivities(SendActivitiesHandler)"/>
+        public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext turnContext, Activity[] activities, CancellationToken cancellationToken)
+        {
+            if (turnContext == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+
+            if (cancellationToken == null)
+            {
+                throw new ArgumentNullException(nameof(cancellationToken));
+            }
+
+            if (activities == null)
+            {
+                throw new ArgumentNullException(nameof(activities));
+            }
+
+            if (activities.Length == 0)
+            {
```

Comment:
We've gone back and forth over this, but I *think* this is a mistake. 

If someone calls this method with no activities, just return "done" (well, an empty ResourceResponse array, anyway).